packages/runtime-core/src/components/KeepAlive.ts

```diff
@@ -261,11 +262,12 @@ const KeepAliveImpl: ComponentOptions = {
 
       // for async components, name check should be based in its loaded
       // inner component if available
-      const name = getComponentName(
-        isAsyncWrapper(vnode)
-          ? (vnode.type as ComponentOptions).__asyncResolved || {}
-          : comp
-      )
+      const name =
+        getComponentName(
+          isAsyncWrapper(vnode)
+            ? (vnode.type as ComponentOptions).__asyncResolved || {}
+            : comp
+        ) || formatComponentName(vnode as any, comp)
```

Comment:
this line doesn't work for async components. 
we should refactor this code like:
```javascript
const _comp = getComponentName(
          isAsyncWrapper(vnode)
            ? (vnode.type as ComponentOptions).__asyncResolved || {}
            : comp
)
const name = getComponentName(comp)||formatComponentName(vnode as any, _comp)
```